ui/components/data-sources/AddDataSourceDialog.js

```diff
@@ -22,7 +22,13 @@ const INITIAL_STATE = {
     validations: {
         name: null,
         type: "warning",
-        options: "success"
+        options: "error",
+
+        // Stores validation states for postgres
+        // properties
+        postgresDetails: {
+            port: "success"
```

Comment:
Why is this one separated? What about the other forms?

Reply:
We only have a default value for the port and want it to be valid without running the validators first.

Reply:
Other forms (in mem) don't have default options (or validation state) yet.